examples/host_camera_server.cc

```diff
@@ -193,28 +188,28 @@ const char *get_device_family()
   const char *device_family = "avfoundation";
 #elif __linux__
   const char *device_family;
-	  device_family = "v4l2";
+          device_family = "v4l2";
 #endif
 
   return device_family;
 }
 
-int init_device_and_input_context(stream_ctx_t *stream_ctx, const char *device_family, const char *device_index, int width, int height, int fps)
+int init_device_and_input_context(stream_ctx_t *stream_ctx, std::string &device_family, std::string &dev_video, int width, int height, int fps)
```

Comment:
You have `dev_video, width, height, fps` at global scope. So, you are tainting global scope variables in this function. I suggest you need to change design. Consider to avoid having these variables at global scope - move them to where you need them (main function?). For device you can do this:
```
std::string dev_video = get_all_dev_nodes();
```
Though you better change function name since it's misleading...

Reply:
updated